app/controllers/reputation_web_service_controller.rb

```diff
@@ -20,80 +20,60 @@ def action_allowed?
+    end
+    peer_review_grade = 100.0 * temp_sum / (weight_sum * max_question_score)
+    peer_review_grade.round(4)
+  end
+
+  def get_peer_reviews_for_responses(reviewer_id, team_id, valid_response)
+    peer_review_grades_list = []
+    valid_response.each do |response|
+      review_grade = calculate_peer_review_grade(response)
+      peer_review_grades_list << [reviewer_id, team_id, review_grade] unless review_grade.nil?
+    end
+    peer_review_grades_list
+  end
+
+  # db query, return peer reviews
+  def get_peer_reviews(assignment_id_list, round_num, has_topic)
     raw_data_array = []
-    assignment_ids = []
-    assignment_ids << assignment_id
-    assignment_ids << another_assignment_id unless another_assignment_id.zero?
-    ReviewResponseMap.where('reviewed_object_id in (?) and calibrate_to = ?', assignment_ids, false).each do |response_map|
+    ReviewResponseMap.where('reviewed_object_id in (?) and calibrate_to = ?', assignment_id_list, false).each do |response_map|
       reviewer = response_map.reviewer.user
       team = AssignmentTeam.find(response_map.reviewee_id)
       topic_condition = ((has_topic && (SignedUpTeam.where(team_id: team.id).first.is_waitlisted == false)) || !has_topic)
       last_valid_response = response_map.response.select { |r| r.round == round_num }.max
       valid_response = [last_valid_response] unless last_valid_response.nil?
-      next unless topic_condition && valid_response && !valid_response.empty?
-
-      valid_response.each do |response|
-        answers = Answer.where(response_id: response.id)
-        max_question_score = begin
-                               answers.first.question.questionnaire.max_question_score
-                             rescue StandardError
-                               1
-                             end
-        temp_sum = 0
-        weight_sum = 0
-        valid_answer = answers.select { |a| (a.question.type == 'Criterion') && a.answer }
-        next if valid_answer.empty?
-
-        valid_answer.each do |answer|
-          temp_sum += answer.answer * answer.question.weight
-          weight_sum += answer.question.weight
-        end
-        peer_review_grade = 100.0 * temp_sum / (weight_sum * max_question_score)
-        raw_data_array << [reviewer.id, team.id, peer_review_grade.round(4)]
+      if (topic_condition == true) && !valid_response.nil? && !valid_response.empty?
+        raw_data_array += get_peer_reviews_for_responses(reviewer.id, team.id, valid_response)
       end
     end
     raw_data_array
   end
 
-  # special db query, return quiz scores
-  def db_query_with_quiz_score(assignment_id, another_assignment_id = 0)
-    raw_data_array = []
-    assignment_ids = []
-    assignment_ids << assignment_id
-    assignment_ids << another_assignment_id unless another_assignment_id.zero?
-    teams = AssignmentTeam.where('parent_id in (?)', assignment_ids)
-    team_ids = []
-    teams.each { |team| team_ids << team.id }
+  def get_ids_list(tables)
```

Comment:
Can you use [map](https://www.geeksforgeeks.org/ruby-array-map-function) function for this?